src/EditorFeatures/Core/Implementation/IntelliSense/Completion2/RoslynCompletionItemSource.cs

```diff
@@ -0,0 +1,202 @@
+
+                var item = Convert(roslynItem, imageIdService, completionService, filterCache, needsCustomCommit);
+                item.Properties.AddProperty(TriggerSnapshot, triggerLocation.Snapshot);
+                return item;
+            });
+
+            return new EditorCompletion.CompletionContext(
+                items,
+                applicableSpan, 
+                useSoftSelection: false, 
+                useSuggestionMode: completionList.SuggestionModeItem != null,
+                suggestionModeDescription: completionList.SuggestionModeItem?.DisplayText);
+        }
+
+        private EditorCompletion.CompletionItem Convert(
+            RoslynCompletionItem roslynItem, 
+            IImageIdService imageService, 
+            CompletionService completionService, 
+            Dictionary<string, CompletionFilter> filterCache,
+            bool needsCustomCommit)
+        {
+            var imageId = imageService.GetImageId(roslynItem.Tags.GetGlyph());
+            var insertionText = roslynItem.DisplayText; // TODO
+            var filters = GetFilters(roslynItem, imageService, filterCache);
+            var item = new EditorCompletion.CompletionItem(
+                roslynItem.DisplayText,
+                this,
+                imageId,
+                filters,
+                suffix: "Test Suffix",
+                useCustomCommit: needsCustomCommit,
+                insertText: insertionText,
+                roslynItem.SortText,
+                roslynItem.FilterText,
+                attributeIcons: ImmutableArray<AccessibleImage>.Empty);
+
+            item.Properties.AddProperty(RoslynItem, roslynItem);
+            return item;
+        }
+
+        private ImmutableArray<CompletionFilter> GetFilters(RoslynCompletionItem item, IImageIdService imageService, Dictionary<string, CompletionFilter> filterCache)
+        {
+            var result = new List<CompletionFilter>();
+            foreach (var filter in CompletionItemFilter.AllFilters)
+            {
+                if (filter.Matches(item))
+                {
+                    if (filterCache.ContainsKey(filter.DisplayText))
+                    {
+                        result.Add(filterCache[filter.DisplayText]);
+                    }
+                    else
+                    {
+                        var itemFilter = new CompletionFilter(filter.DisplayText, filter.AccessKey.ToString(), imageService.GetImageId(filter.Tags.GetGlyph()));
+                        filterCache[filter.DisplayText] = itemFilter;
+                        result.Add(itemFilter);
+                    }
+                }
+            }
+
+            return result.ToImmutableArray();
+        }
+
+        public Task<object> GetDescriptionAsync(EditorCompletion.CompletionItem item, CancellationToken cancellationToken)
+        {
+            return Task.FromResult<object>("Documentation!"); // Editor doesn't call this yet
```

Comment:
We call this method. Please use `*Element` classes in namespace `Microsoft.VisualStudio.Text.Adornments` from the Text.UI assembly to construct cross-platform UI.
QuickInfo recently started consuming these APIs.